crates/wasm/scripts/build.sh

```diff
@@ -1 +1 @@
-wasm-pack build --debug --scope swc -t nodejs
+wasm-pack build --debug --scope swc -t nodejs --features plugin
```

Comment:
@kdy1 if I understood correctly does it have lot meaning to be able to compile `wasm` target natively? most of code which actually will use `swc` (`swc/plugin`) will be guarded by wasm build target and it'll be mostly just noop.

Reply:
>this is part I'm not fully getting now: does build script to wasm need to opt in features manually for https://github.com/swc-project/swc/pull/4035/files#diff-5b743c71dfc992501e49511ae0f681b46af76bac40bec59caf09667b4dd2b12cR19?

Yes, otherwise it breaks cargo operations.

> if that's the case, essentially wasm package need to be written to be able to compile with native target which doesn't do thing and then actual codebase for wasm target?

Maybe, but I think enabling `wasmer/default` `wasmer-wasi/default` from `devDependencies` would be enough

Reply:
> if I understood correctly does it have lot meaning to be able to compile wasm target natively? 


`buildable-by-default` is important for development.


> most of code which actually will use swc (swc/plugin) will be guarded by wasm build target and it'll be mostly just noop.

I didn't get it

Reply:
> Maybe, but I think enabling wasmer/default wasmer-wasi/default from devDependencies would be enough

the main question I have is this:

```
//wasm/lib.rs

#[target(not(wasm32))]
fn transform_xx() {
  // noop since we can't refer swc, which depends on swc/plugin relies on plugin_runner's wasm32 target path
}

#[target(not(wasm32))]
fn transform_xx() {
  // actual code
}
```

`cargo build`, and `cargo check` will work but it's nearly noop as it won't be able to check any actual code under right target we need to build. The part I'm curious is if it's worth to try to keep `crates/wasm`  to be built in this way, while it only spends build time. Any reason not to exclude it as https://github.com/swc-project/swc/pull/4012 attempts?

Reply:
The important command is `cargo check` and `cargo test`.
But `cargo check` is important because of ide support, and `rust-analyzer` uses `cargo check` with test config, so it follows `devDependencies`.
`cargo test` also uses `devDependencies`, so it would be fine.

Reply:
It looks like my point wasn't clearly delivered 😅 but it won't block immediate work at least. When I have some time I'll try to split this topic into different one for better claritity. Anyway, for now I'll proceed PR with suggested approaches.